src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -330,6 +332,7 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		this._currentCommand.commandStartMarker = options?.marker || this._currentCommand.commandStartMarker;
 		if (this._currentCommand.commandStartMarker?.line === this._terminal.buffer.active.cursorY) {
 			this._currentCommand.commandStartX = this._terminal.buffer.active.cursorX;
+			this._onCommandStartChanged.fire(this._currentCommand as any as ITerminalCommand);
```

Comment:
should we just change the type of event to be `| ICurrentPartialCommand` to avoid this cast and to make it clearer?

Reply:
https://github.com/microsoft/vscode/pull/241493